src/Workspaces/Core/Portable/EmbeddedLanguages/Json/JsonLexer.cs

```diff
@@ -0,0 +1,413 @@
+            }
+            else if (IsAt("\r") || IsAt("\n"))
+            {
+                Position++;
+                return CreateTrivia(JsonKind.EndOfLineTrivia, GetSubPattern(start, Position));
+            }
+
+            return null;
+        }
+
+        public JsonTrivia? ScanComment()
+        {
+            if (IsAt("//"))
+            {
+                return ScanSingleLineComment();
+            }
+            else if (IsAt("/*"))
+            {
+                return ScanMultiLineComment();
+            }
+            else if (IsAt("/"))
+            {
+                var start = Position;
+                Position++;
+
+                var chars = GetSubPattern(start, Position);
+                return CreateTrivia(JsonKind.SingleLineCommentTrivia, chars,
+                    ImmutableArray.Create(new EmbeddedDiagnostic(
+                        WorkspacesResources.Error_parsing_comment,
+                        GetSpan(chars))));
+            }
+
+            return null;
+        }
+
+        private JsonTrivia ScanSingleLineComment()
+        {
+            var start = Position;
+            Position += 2;
+
+            while (Position < Text.Length && 
+                   this.CurrentChar is var ch &&
+                   ch != '\r' && ch != '\n')
+            {
+                Position++;
+            }
+
+            var chars = GetSubPattern(start, Position);
+            if (Position == start + 2)
+            {
+                var diagnostics = ImmutableArray.Create(new EmbeddedDiagnostic(
+                    WorkspacesResources.Unterminated_comment,
+                    GetSpan(chars)));
+                return CreateTrivia(JsonKind.SingleLineCommentTrivia, chars, diagnostics);
+            }
+
+            return CreateTrivia(JsonKind.SingleLineCommentTrivia, chars);
+        }
+
+        private JsonTrivia ScanMultiLineComment()
+        {
+            var start = Position;
+            Position += 2;
+
+            while (Position < Text.Length &&
+                   !IsAt("*/"))
+            {
+                Position++;
+            }
+
+            if (IsAt("*/"))
+            {
+                Position += 2;
+                return CreateTrivia(JsonKind.MultiLineCommentTrivia, GetSubPattern(start, Position));
+            }
+
+            Debug.Assert(Position == Text.Length);
+            return CreateTrivia(JsonKind.MultiLineCommentTrivia, GetSubPattern(start, Position),
+                ImmutableArray.Create(new EmbeddedDiagnostic(
+                    WorkspacesResources.Unterminated_comment,
+                    GetTextSpan(start, Position))));
+        }
+
+        public TextSpan GetTextSpan(int startInclusive, int endExclusive)
+            => TextSpan.FromBounds(Text[startInclusive].Span.Start, Text[endExclusive - 1].Span.End);
+
+        public bool IsAt(string val)
+            => TextAt(this.Position, val);
+
+        private bool TextAt(int position, string val)
```

Comment:
This method is only used in `IsAt`. Could be inlined?

Reply:
I'm ok with it as is.  #won'tfix